lib/core/utils/preload-cssom.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Returns a then(able) queue of CSSStyleSheet(s)
+ * @param {Object} ownerDocument document object to be inspected for stylesheets
+ * @param {number} timeout on network request for stylesheet that need to be externally fetched
+ * @param {Function} getSheetFromTextFn a utility function to generate a style sheet from text
+ * @return {Object} queue
+ * @private
+ */
+function loadCssom(ownerDocument, timeout, getSheetFromTextFn) {
+	const q = axe.utils.queue();
+
+	Array.from(ownerDocument.styleSheets).forEach(sheet => {
+		if (sheet.disabled) {
```

Comment:
You should add `&& sheet.cssRules.length <= 0`, rather than add an if statement in the try/catch block. Currently the code looks like it's missing an "else".

Reply:
we cannot move .cssRules to here, as we need the catch to trigger for external stylesheets. Trying to read a .cssRules on external resource throws a SecurityError, which flows into the catch block. This is documented below.

Reply:
Although I have refactored slightly